.github/workflows/tests.yml

```diff
@@ -301,8 +354,10 @@ jobs:
 
   trial-olddeps:
     # Note: sqlite only; no postgres
-    if: ${{ !cancelled() && !failure() }} # Allow previous steps to be skipped, but not fail
-    needs: linting-done
+    if: ${{ !cancelled() && !failure() && needs.changes.outputs.trial == 'true' }} # Allow previous steps to be skipped, but not fail
```

Comment:
Aside: I _think_ the olddeps job doesn't care if poetry.lock changes, only if pyproject.toml does. Not worth the yaml faff to encode that here though!